modules/ingress-certificates.adoc

```diff
@@ -0,0 +1,120 @@
+controller puts the certificate in a secret named `metrics-tls` in the
+`openshift-ingress-operator` namespace. Additionally, the Ingress Operator
+requests a certificate for each Ingress Controller, and the `service-ca`
+controller puts the certificate in a secret named `router-metrics-certs-<name>`,
+where `<name>` is the name of the Ingress Controller, in the
+`openshift-ingress` namespace.
+
+Each Ingress Controller has a default certificate that the Controller uses for
+secured routes that do not specify their own certificates. You can set a custom
+default certificate, or else the Operator generates a certificate for you. The
+Operator uses its own self-signed signing certificate to sign any default
+certificate that it generates. The Operator generates this signing certificate
+and puts it in a secret named router-ca in the `openshift-ingress-operator`
+namespace. When the Operator generates a default certificate, it puts the
+default certificate in a secret named `router-certs-<name>` (where `<name>` is the
+name of the Ingress Controller) in the `openshift-ingress` namespace.
+
+[WARNING]
+====
+The Ingress Operator generates a default certificate for an Ingress Controller
+to serve as a placeholder until you configure a custom default certificate. Do
+not use operator-generated default certificates in production clusters.
+====
+
+[discrete]
+== Workflow
+
+image::ingress-certificates-workflow.png[Ingress certificate workflow]
+
+.Default certificate workflow
+
+image:darkcircle-0.png[20,20] An empty `defaultCertificate` field causes the Ingress Operator to use its self-signed CA to generate a serving certificate for the specified domain.
+
+image:darkcircle-1.png[20,20] The default CA certificate and key generated by the Ingress Operator. Used to sign operator-generated default serving certificates.
+
+image:darkcircle-2.png[20,20] The wildcard default serving certificate, created by the Ingress Operator and signed using the generated default CA certificate.
+
+image:darkcircle-3.png[20,20] The router deployment. Uses the certificate in `secrets/router-certs-default` as its default front-end server certificate.
+
+image:darkcircle-4.png[20,20] The contents of the wildcard default serving certificate (public and private parts) are copied here to enable OAuth integration.
+
+image:darkcircle-5.png[20,20] Transitional resource containing the certificate (public part) of the Operator-generated default CA certificate; read by OAuth and the web console to establish trust. This object will be removed in a future release.
+
+image:darkcircle-6.png[20,20] The public (certificate) part of the default serving certificate. Replaces the `configmaps/router-ca` resource.
+
+image:darkcircle-7.png[20,20] The user updates the cluster proxy configuration with the CA certificate that signed the `ingresscontroller` serving certificate. This enables components like auth, the web console, and the registry to trust the serving certificate.
```

Comment:
This item should be moved to "Custom certificate workflow".  Updating the cluster proxy configuration is part of the custom workflow, not the default workflow.

A note on terminology:  When we talk about the "default certificate", we are referring to _the serving certificate that the ingress controller uses by default when a route does not specify a serving certificate_.  The default certificate can either be generated by the operator or provided by the user, so in order to avoid saying "default default certificate", I have been referring to the former as an "operator-generated default certificate" and the latter as a "custom default certificate".  I am not sure what convention we are using (or should be using) in product documentation, but in any case, we need to be consistent and careful to avoid confusion.

I have some comments regarding the diagram specifically; because GitHub does not allow me to comment on the graphic, I will make my comments here:

In `ingress-certificates-workflow.png`, the "User" box should be shaded green to denote that it is part of the custom workflow.  (It is not necessary for the user to update the cluster proxy configuration as part of the default workflow.)

The diagram shows two instances of `deployments/router-default` with `secrets/router-certs-default`, both shaded blue to indicate that they are part of the default workflow.  We need one instance for the default workflow and one for the custom workflow.  I suggest changing the lower instance of `deployments/router-default` and `secrets/router-certs-default` as follows:

* Shade the boxes green to denote that they are part of the custom workflow.
* Change the name of the secret to `secrets/router-certs-custom` to denote that the deployment is using a custom default certificate.
* Change the notes for `deployments/router-default` and `secrets/router-certs-custom` (neé `secrets/router-certs-default`) to indicate that the deployment is using a custom default certificate and that the secret was created by the user.

Similarly, the diagram shows two instances of `OAuth` and two instances of `Web console` (or `web console`, with a small "w"; see next paragraph) that are all blue.  The `OAuth` and `Web console` boxes that have `configmaps/proxy-ca` boxes inside, and the `configmaps/proxy-ca` boxes themselves, should be all shaded green, and their notes should be changed to reflect that they are using the user-provided CA certificate from the cluster proxy configuration.

One instance of `Web console` has `web console` with a small "w"; it should be capitalized for consistency.  The `OAuth` and `Web console` labels of the boxes on the upper-right use a proportionally spaced type face, and the `OAuth` and `Web console` labels of the boxes on the bottom-left use a monospaced type face; they should be consistent, probably all proportionally spaced to denote that the boxes represent abstract actors and not concrete API objects.

Reply:
@Miciah Thank you! I am shared your comments with the Brand team so that they can help us update the diagram.

Reply:
@Miciah The PR is now updated with a revised diagram. PTAL and let me know if further edits are needed. Thanks again!